libs/seiscomp/math/filter/iirdifferentiate.cpp

```diff
@@ -30,15 +30,13 @@ namespace Filtering
 
 template <typename T>
 IIRDifferentiate<T>::IIRDifferentiate(double fsamp) : _fsamp(fsamp) {
-	reset();
 	setSamplingFrequency(fsamp);
 }
 
 
 template <typename T>
 IIRDifferentiate<T>::IIRDifferentiate(const IIRDifferentiate<T> &other) {
-	_fsamp = 0.0;
-	reset();
+  setSamplingFrequency(other._fsamp);
```

Comment:
If the compiler initializes _fsamp with 0.0 then this function does nothing and the other reset states are still unconfigured. The original two lines stable in that regard.